gremlin-server/src/main/java/org/apache/tinkerpop/gremlin/server/handler/StateKey.java

```diff
@@ -57,4 +58,9 @@ private StateKey() {}
      * The key for the current request.
      */
     public static final AttributeKey<RequestMessage> REQUEST_MESSAGE = AttributeKey.valueOf("request");
+
+    /**
+     * The key for the current {@link AuthenticatedUser}.
+     */
+    public static final AttributeKey<AuthenticatedUser> AUTHENTICATED_USER = AttributeKey.valueOf("authenticatedUser");
```

Comment:
nice. looks like this body of work will also close:

https://issues.apache.org/jira/browse/TINKERPOP-2418

We will need to remember to close that issue too when this merges

Reply:
There is a warning in the changelog you cannot miss before the release.